### ISSUE TITLE
SUGGESTION for more MINI-MODS - batteries for storing FE

### ISSUE THREAD

**Issue by markygnlg** [canitzp/SolarHelmet#2]
I've been using your mod because it is VERY configurable & small (for the mod "[DireGoo](https://www.curseforge.com/minecraft/mc-mods/diregoo)") - unfortunately the 'turret' item does not store a charge, whereas the 'Goo Remover' gun does store a charge.

I ended up adding "[Solar Generation](https://www.curseforge.com/minecraft/mc-mods/solargeneration)" mod but it is not configurable & I need to multi-step-recipe my way to a decent FE storage solution.

Could you make a battery 'system' (ie. collect/store/transfer FE) from this helmet to a battery & then into something else?

I added the 2 mods as examples because the 'diregoo' mod has items than can & can't store FE (ie. Goo Remover' gun VS 'turret') & the 'Solar Generation' mod stores FE in its solar panels, even after breaking.

'Solar Generation' also has the option of making solar helmets & also the option of converting solar panels into helmets (I do not know if the stored FE in kept in the conversion) - but it's one of those unconfigurable-multi-step-recipe mods where all I do is make a batch, re-make a batch, search inventory for more recipe ingredients, make another batch, re-make a batch, search inventory for more recipe items........... I WANT A MINI-MOD instead!

A battery mini-mod for storage & the ability of transferring of FE via a mods like "[Refined Pipes](https://www.curseforge.com/minecraft/mc-mods/refined-pipes)" or "[Pretty Pipes](https://www.curseforge.com/minecraft/mc-mods/pretty-pipes)" would be awesome!!

Thanks for this mod & your feeder helmet mod too!!.

**Comment by canitzp** [comment]
I'm very sorry for the late response, I haven't seen this issue yet. Of course your idea is great and a Mini-Mod for just a battery is a great idea! I think I call it 'JustABattery'.
Do you know some guy who can make me a texture for a battery? @markygnlg

**Comment by canitzp** [comment]
> & the ability of transferring of FE via a mods like "Refined Pipes" or "Pretty Pipes" would be awesome!!

I'm not fully sure what actually is your intention? Do you meant that I should do a energy transfer mod, like pretty pipes is for items? Because that wouldn't be a mini mod, it involves a lot of codebase.

**Comment by markygnlg** [comment]
Hi,
Thanks for the question! I meant EXACTLY as you said:

> a Mini-Mod for just a battery is a great idea! I think I call it 'JustABattery'.
.. just to store/transfer energy (like the way "Refined Pipes" or "Pretty Pipes" does, I think - I haven't used them but youtube videos show they can).


> Do you know some guy who can make me a texture for a battery? @markygnlg
.. No, sorry I do not.

**Comment by markygnlg** [comment]
> > & the ability of transferring of FE via a mods like "Refined Pipes" or "Pretty Pipes" would be awesome!!
> 
> I'm not fully sure what actually is your intention? Do you meant that I should do a energy transfer mod, like pretty pipes is for items? Because that wouldn't be a mini mod, it involves a lot of codebase.

NEVER!!! I just want mini mods - in every game & in every circumstance! If I want to feel like I've done something to be proud of I'll go for a jog, swim, even a 2hr walk!! I'd LOVE to learn to make mods too!! When I play a game I just want to have fun & add mods to take away some of the grind.

In my personal opinion there are 'big' mods that are great & there are 'small' mods that are great - then there're the 'mid sized' mods, where some things seem utterly useless &/or unnecessary but they add to the load time & memory/CPU/GPU requirements.

I LOVE your mini-mod ideas! I wish there was a mini-mod section/tag on every mod website!
Thanks for your great work!

**Comment by canitzp** [comment]
So I started with JustABattery and I now have to think of the capacity. I either can set it to a fixed value (eg 100kFE), which is configurable by config or I'm thinking of a generic way, where you would craft the battery with an initial capacity and the upgrade it (with easy craftable items) to higher capacities and transfer rates. Is this second though to much?

**Comment by markygnlg** [comment]
It's up to you - you're the genius - but my 2 cents are : I LOVE your mods coz they're super configurable.

Maybe:
- 'JustABattery'+'JustABattery' = **JustABatteryDoubleSize**
- 'JustABattery'+'JustABattery'+'JustABattery' = **JustABatteryTripleSize**
- 'JustABattery'+'JustABattery'+'JustABattery'+'JustABattery' = **JustABatteryQuadSize**
- 'JustABattery'+'JustABattery'+'JustABattery'+'JustABattery'+'JustABattery' = **JustABatteryQuintSize**

Or:
- 'JustABattery' + 'JustABattery' = **JustABatteryDoubleSize**
- JustABatteryDoubleSize + JustABatteryDoubleSize = **JustABatteryTripleSize**
- JustABatteryTripleSize + JustABatteryTripleSize = **JustABatteryQuadSize**
- JustABatteryQuadSize + JustABatteryQuadSize = **JustABatteryQuintSize**

in any of those two scenarios a person can configure the 'minimum' battery size & 'artificially' create their own customized **'grind'** setting(s) ?

**Comment by canitzp** [comment]
Mhh, I think you're right, but I combine my idea and yours.
Instead of just do batter+batter=double battery, I also consider another factor.
- battery+battery=dual battery pack
You can combine them as you want, like single+single=double or double+single=triple, or triple+double=quint

The other factor is the transfer rate. Combining batteries increases their capacity, but to increase the transfer rate you need to add metal (probably wire, but i don't want additional items).
This would follow the same principal, so single+single=double, and so on.

**Comment by markygnlg** [comment]
> ...I either can set it to a fixed value (eg 100kFE), which is configurable by config or I'm thinking of a generic way, where you would craft the battery with an initial capacity and the upgrade it (with easy craftable items) to higher capacities and transfer rates...

maybe you can make whats simpler for you then wait for people to reply on the curseforge page  for a better understanding of what your fans want/need - you can then decide to upgrade your mod with your fans requests or to spin-off into a seperate mod

**Comment by canitzp** [comment]
@markygnlg can we continue writing over discord? Since you're a good help for my brain, you may want to also test JustABattery, once I have something to show and also give me other ideas.
My dc tag is canitzp#6125

**Comment by markygnlg** [comment]
To be honest, I havn't played mc in a few weeks - maybe 2 months.

2nd, I went back to 1.12 long before then because I wanted to try out those famous & popular mods I kept hearing about.

3rd, I don't have discord - I hate the way social media has been taking up so much of my time so when I'd lost over 20kg, social media became less & less of good thing & more of a poison - remember, I'm not a programmer but if I was I'd find social media rewarding but, as it stands, I find walking more rewarding since the internet helps me not at all (for the most part, however, helping you is GREAT)

4th, you may want to re-read my posts to you since I've been editing them too a CRAZY level to be clear as I can coz I REALLY, REALLY, REALLY, REALLY, REALLY, REALLY, REALLY LOVE mini-mods - less load time/CPU/GPU/etc
...

keep up the great work but make sure you like it. I've met mechanics, PC repair guys, carpenters, programmers, etc, that loved their hobby only to find no happiness &/or money &/or rewarding feelings, etc when they get a job that involves their hobby - do it coz you like it, u don't gotta love it, but do it if you like it (even if it means screen recording the 'what/where/why/how' you do the things you do & then posting it onto YouTube to help newbies - it's not just about mods but it can also be about being a tutor to give you that great rewarding experience).

Like I said, I LOVE mini-mods & you're the genius. Trust yourself, trust your fans & don't make life harder than it needs to be.

**Comment by canitzp** [comment]
I don't play Minecraft often too. I don't really find the time to and when I have time, I rather focus on other games or programming my mods.
I understand why you don't have discord/any social media stuff.

I had a little problem over the last years, where I never had the mood to actually start developing big mods. You can see a lot of failed projects here on github. But I gonna change my job next year and so hope the get more mood (currently my job I software dev too, but I change it). This is why I started with Mini-Mods to begin with, because they are easily maintainable and fast to write.

Do you have any stuff, like modpacks on curseforge?

**Comment by markygnlg** [comment]
> Do you have any stuff, like modpacks on curseforge?

No. Since moving interstate & changing jobs I only use games to pass the time - which is extremely rare. From my understanding "Marriage, Mortgage, Multiply" ("The 3 M's") seems to make most people 'feel happy' [I prefer 'feeling content'].
...Then there's health - weight loss, quitting smoking/drinking, better friends, regular sleep...

> I had a little problem over the last years, where I never had the mood to actually start developing big mods.

Think about that & the next few things you said - I'm not here to tell you things, I'm no expert. I'd write books if I was. - your mood, most likely, won't change: either you like it or you won't. Only if you like it, then your mood MIGHT change - but you mentioned "the mood ... developing big mods."

You can see a lot of failed projects here on github

You have no 'failed' projects - you either got bored with them &/or people had no reason to use them &/or people did not find them - here are soooooooooooooooo many now.

But I gonna change my job next year and so hope the get more mood 

new job does not mean more mood. It's not a bad thing. I would have not lost 20kg if I kept to the same routine.

This is why I started with Mini-Mods to begin with, because they are easily maintainable and fast to write.

...general thing I've learned - the more ppl that like something then the more people agree with something. Movies, music, religions - who's right? Who's wrong? Is there a wrong style of music, TV shows, movies, religion, cars, artwork?
You like mini-mods, for your several reasons, I like mini-mods for several (different) reasons. I know I'm not wrong, I know you're not wrong...

Like I said, I LOVE mini-mods & you're the genius. Trust yourself, trust your fans & don't make life harder than it needs to be. Also,
do it coz you like it, u don't gotta love it, but do it if you like it - even if it means screen recording the 'what/where/why/how' you do the things you do & then posting it onto YouTube to help others - if you like me helping you with ideas, imagine youself helping others

**Comment by markygnlg** [comment]
on a lighter note

> The other factor is the transfer rate. Combining batteries increases their capacity, but to increase the transfer rate you need to add metal (probably wire, but i don't want additional items).

OOps, I forgot about transfer rate... you're right! As always

**Comment by markygnlg** [comment]
Oh! Let's not forget that v1.17 is out now & has new resources! (copper n stuff?)

**Comment by canitzp** [comment]
> Oh! Let's not forget that v1.17 is out now & has new resources! (copper n stuff?)

Oh yeah i completely forgot copper XD. Thats a nice source for the transfer rate increase.

When I say I don't have the mood to develop big mods in my free time, I really ment that I don't have the mood to develop at all. This is only due to my current job as software dev. I didn't has this before I got the job and so I change my department from software dev to system engineering. This has less to do with developing and so I can spend the free time I get on my long time hobby, prorgamming mods.
For over 6 years now I develop mods for Minecraft and it is the reason I now do what I do and why I have learned this job (in germany you learn a profession for multiple years, instead of just working as someting).

You really sound like a mental health person and I like the conversation :D

**Comment by markygnlg** [comment]
> When I say I don't have the mood to develop big mods in my free time, I really ment that I don't have the mood to develop at all.

#**1st copy/paste:** your mood, most likely, won't change: either you like it or you won't. Only if you like it, then your mood MIGHT change
#**2nd copy/paste:** I've met mechanics, PC repair guys, carpenters, programmers, etc, that loved their hobby only to find no happiness &/or money &/or rewarding feelings, etc when they get a job that involves their hobby

... Sorry, I meant no offence by the "copy/paste" comments - they're meant to show/display to you that you **ARE** growing up, your feelings are normal - ie. Do you believe in Santa Clause? Easter Bunny? Tooth Fairies? Brand named supercars? Brand named shoes/clothes? Do you have posters of supermodels/actresses/bands on your walls?

.. **more importantly, the 2nd seems most relevant to you** for now - "the older, the wiser"; your 'mood' **WILL** change when you get older & wiser; do you **NOW** like - Stamp collecting? Homebrew wine/beer? Gardening? Looking after the Grand kids? No, that comes later.


> This is only due to my current job as software dev. I didn't has this before I got the job and so I change my department from software dev to system engineering. This has less to do with developing and so I can spend the free time I get on my long time hobby, prorgamming mods.

#1 'free time' does not mean you (always/sometimes) do your 'hobby' - 'time poor' is real, the more kids/ programming/ work/ travel/ responsibilities = less time;
#2 'long time hobby' is just a 'hobby' ... very important? not important? sometimes ?


> in germany you learn a profession for multiple years, instead of just working as someting).

- reminds me of why Russian hackers/programmers are now 'famous' ('in-famous') for ransomware. They were compared with 'experts' in the Eighties & early nineties ("you learn a profession for multiple years, instead of just working as someting") but now ... ? 

> You really sound like a mental health person and I like the conversation :D

I've met many, many,many, many,many, many,many, many,many, many, ppl in my life which means I've had many, many,many, many, 'experiences' - via 1st person, 3rd person interactions & whatever else there is in between - I'm tall, blonde, green eyed & have a deep voice (Polish background/heritage) so where-ever I've lived I've ALWAYS fitted in - ie. either 'western' or 'European' communities/suburbs (especially after I realised that being 'tall, blonde, green eyed & deep voiced' is valued in ALL western/European societies!!)

You're German, you're **Extremely** smart (only smart ppl are smart enough to know that there is alot of information that they don't know, & therefore will convince themselves that they are not smart!!)

you're smart, & thats why you **know** that you're not. Only because we know ther is much, much, much more information in the world that we do not know. Dumb ppl don't know this, they're aren't smart enough to include that information in their brain... if your friends/family/co-workers are beneath your intelligence then **you WILL FEEL DUMB & a failure** - pity others, not yourself - you're smart. Very smart. It takes a loooooooooooooong time to accept that. It's normal.

**Comment by markygnlg** [comment]
I'm thinking that you're close to 24years old but you comment of "I change my department from software dev to system engineering. This has less to do with developing and so I can spend the free time I get on my long time hobby" makes me think that you're about 33years old - either way, you're changing to improve your life: you **KNOW** that an easier job is ... ? easier? better? rewarding? money? time? health?...

**Comment by canitzp** [comment]
I change my job because I don't like the current "working climate" and because I didn't get the pay out I want (My new job gives me 76% more money and others which I declined where in the same region, much more than what I get today). I also like my current tasks about 50% and I hope, that the new job increases this number. (Also they have some more nice things, which I don't all write down here)
I'm actually 22 years old and I have never worked in another company, but I want to take this journey, to find the most suitable job for me. Programming is a long time (6+ years OOP with Java and more experience before that with some ti-basics, which lead me to this (software world) in the first place) and very important hobby for me. Most of my now friends, I met because I started to develop and explored new communities. Every single day I check if my mods have some issues or if something happened, there is no day without it. And this is why I understand you, who don't want such a life, someday I also may have enough of this "always online" lifestyle.

**Comment by canitzp** [comment]
I just wanna share this picture, as you can see, there are already batteries with charges. Currently the capacity maximum is 2.52 million FE at level 126, but I probably increase this even further. The transfer rate increase is not implemented yet.
![2021-07-29_15 21 50](https://user-images.githubusercontent.com/12819060/127499496-2510ef15-f5c2-4615-9f0b-4c21cf4a9082.png)

**Comment by canitzp** [comment]
@markygnlg Here it is alive: https://www.curseforge.com/minecraft/mc-mods/justabattery

**Comment by markygnlg** [comment]
Not found
We were unable to find the page or file you were looking for.

I'm guessing it takes time for curseforge to accept it? I've heard that curseforge has issues when updating mods...

BTW, rekn I'm gonna reinstall mc since I can't find a game that I can get into - every game seems to get too detailed at the start where I need to learn too much or it's too slow-started at the beginning. You're mini-mods are the reason I'm gonna retry it!

**Comment by markygnlg** [comment]
P.S.
I have another idea for a mini-ish-mod: an OP redstone tool & armour set such as:
[Redstone-Tools [v1.15 & v1.16]](https://www.curseforge.com/minecraft/mc-mods/redstone-mod-by-poetic_d) or [Redstone Gear Mod](https://www.curseforge.com/minecraft/mc-mods/redstone-gear-mod-1-15-2) [v1.14 but also works on v1.15]

 The reason for it being redstone is that you need wood pickaxe -> stone pickaxe -> iron pickaxe to get down to the redstone levels. Maybe a 1.12 version too as tooooo many OP tools &/or armour mods for 1.12 have waaaaaaaaaaaayyyyyyyyyyyyy toooo much junk (but I can't rememeber if 1.12 has redstone).

ATM, I need to edit the mods.toml file in Redstone-Tools to get it to work on 1.16 newer versions, I also discovered that the Redstone Gear Mod for v1.14 worked on v1.15 - so when I posted that the mod's author changed the description to it being a 1.15 version too since no one tests mods, they just quickly ask for an updated version - as I'm sure you know!

TL:DR: OP redstone armour & tools mini-mod for 1.12 to current!! (including 1.14 - 1.16 since ppl want one mod rather than searching multiple mods by multiple mod makers)

**Comment by markygnlg** [comment]
**OH WOW!!!** I just found your github for JustABattery (since curseforge doesn't have it yet) - & you seem to have included what I wanted - but I didn't want to ask you for because I thought it would be too much trouble for you; ie. [Big Buckets](https://www.curseforge.com/minecraft/mc-mods/bigbuckets), where...
>   You can combine them as you want, like single+single=double or double+single=triple, or triple+double=quint

but with that mod EVERY result/upgrade had the EXACT picture: so a 7 Bucket looked EXACTLY like every other 2x, 10x, 29x modded bucket with no way to determine (understandably) if a half filled bucket was a 15 of X30 sized 'upgrade' or a 2 of X4 sized 'upgrade' - it seems like you found a way to do it!

Can't wait; LITERALLY - I've just installed 1.16.5, & now I'll about to install 1.17.x (since I've seen the github page's recipies!!)

**Comment by markygnlg** [comment]
...installed mc1.17.x;

...installed jdk-16.0.2_windows-x64_bin.exe
 ---> BUILD SUCCESSFUL in 23m 2s

...installed forge-1.17.1-37.0.17-installer;

Here We Go!!!!!

**Comment by markygnlg** [comment]
> Here We Go!!!!!

[**No we Don't** - coz compiling is not the same as installing...]

...

...close the launcher (**YES!!** I remembered that fact BEFORE starting mc w/forge);

...try to remember if .mc is in AppData\Local, LocalLow or Roaming (click on stuff until..);

...YAY!! Now enter mods folder (but NO - coz I gotta go & find the compiled mod file 1st!!);

... **THERE** it is!! \JustABattery-master\build\libs\JustABattery-1.0.0.jar ;

...right-click -> copy; go & re-find the .mc/mods folder;

...**OOPS!!** No I don't, there's no mods folder in a fresh install!! [create it...];

**Here We Go v2.00 !!**

**Comment by markygnlg** [comment]
with no other mods installed...

**Comment by canitzp** [comment]
Nice to hear you're happy 😊.
Yeah cursforge is still "reviewing", because it is a new mod and not just an update 🧐. This may take up to a week. Before overwolf took them, this only took several hours at most.
Yeah different textures for different level is a thing, but only up to level 5, all above use the same (level 5) texture. I think about how to display higher tiers, maybe different energy color or so, but is is also a lot of work to create these textures, even tho they are only copy&paste. Higher "tracewidth" isn't shown at all, I just don't know how, without writing a custom renderer.

**Comment by markygnlg** [comment]
Ye, thats why my 2 cents were only worth 2 cents - I was thinking things that was more 'icons' limited which meant less 'result' - like 'Big Buckets (& other mods using 'colors-with-meaning' - nut different mods had different 'meanings')...

..that's why I think that 'make-it-simple-&-wait-for-your-fans-2-tell-ya-WHAT-YOU-NEED-2-BE-POPULAR' - like the way youtube ppl become successsful; patience + adaptation = [potential] success.

...

[OHHHHH, here I gotta link to this guy - I can't tell you why, coz I don't know why ... **ONLY CLICK IF YOU EAT MEAT - - https://www.youtube.com/c/GEORGYKAVKAZLife/featured]

**Comment by markygnlg** [comment]
> Nice to hear you're happy .
> Yeah cursforge is still "reviewing", because it is a new mod and not just an update . This may take up to a week. Before overwolf took them, this only took several hours at most.
> Yeah different textures for different level is a thing, but only up to level 5, all above use the same (level 5) texture. I think about how to display higher tiers, maybe different energy color or so, but is is also a lot of work to create these textures, even tho they are only copy&paste. Higher "tracewidth" isn't shown at all, I just don't know how, without writing a custom renderer.

->like I said,
I L O V E your MINI-MODS soooooooooooo much that;:
Can't wait - LITERALLY!! So I had to...

- install 1.16.5
- install 1.17.x (after seeing your github recipies!!)
- download JustABattery-master.zip;
- install jdk-16.0.2_windows-x64_bin.exe
  --> gradlew build
  --> BUILD SUCCESSFUL in 23m 2s;
- install forge-1.17.1-37.0.17-installer;



Here We Go!!
..


OH!! --> compiling is not the same as installing, so..
- close the launcher (YES!! I remembered that fact BEFORE starting mc w/forge);
- now I'm trying to remember if .mc is isstalled into AppData\Local, LocalLow or Roaming (click on stuff until..);
- YAY!! Now enter mods folder (but NO - coz I gotta go & find the compiled mod file!!? but where is it!??!);
- THERE it is!! \JustABattery-master\build\libs\JustABattery-1.0.0.jar ;
- mouse right-click -> copy; then i gotta go & re-find the .mc/mods folder;
- OOPS!! No I don't, there's no mods folder in a fresh install!! [create it...];
..
..
Here We Go v2.00 !!

...but there're no other mods installed...
... Ohh, Bollocks!

**Comment by canitzp** [comment]
since JustABattery exists now, I gonna close this

**Comment by markygnlg** [comment]
Thanks for the 1.16 version!! Took me a few days to realise that I had no use for a 1.17 version. Take care.
